src/contracts/client/mailboxclient.cairo

```diff
@@ -0,0 +1,122 @@
+#[starknet::contract]
+mod mailboxclient {
+    use alexandria_bytes::{Bytes, BytesTrait, BytesStore};
+    use hyperlane_starknet::interfaces::{
+        IMailbox, IMailboxDispatcher, IMailboxDispatcherTrait, IInterchainSecurityModuleDispatcher,
+        IInterchainSecurityModuleDispatcherTrait, IMailboxClient,
+    };
+    use openzeppelin::access::ownable::OwnableComponent;
+    use openzeppelin::upgrades::{interface::IUpgradeable, upgradeable::UpgradeableComponent};
+    use starknet::{ContractAddress, contract_address_const, ClassHash};
+
+    component!(path: OwnableComponent, storage: ownable, event: OwnableEvent);
+    component!(path: UpgradeableComponent, storage: upgradeable, event: UpgradeableEvent);
+    #[abi(embed_v0)]
+    impl OwnableImpl = OwnableComponent::OwnableImpl<ContractState>;
+    impl OwnableInternalImpl = OwnableComponent::InternalImpl<ContractState>;
+    impl UpgradeableInternalImpl = UpgradeableComponent::InternalImpl<ContractState>;
+
+
+    #[storage]
+    struct Storage {
+        mailbox: ContractAddress,
+        local_domain: u32,
+        hook: ContractAddress,
+        interchain_security_module: ContractAddress,
+        #[substorage(v0)]
+        ownable: OwnableComponent::Storage,
+        #[substorage(v0)]
+        upgradeable: UpgradeableComponent::Storage,
+    }
+
+    #[event]
+    #[derive(Drop, starknet::Event)]
+    enum Event {
+        #[flat]
+        OwnableEvent: OwnableComponent::Event,
+        #[flat]
+        UpgradeableEvent: UpgradeableComponent::Event,
+    }
+
+
+    #[constructor]
+    fn constructor(ref self: ContractState, _mailbox: ContractAddress, _owner: ContractAddress) {
+        self.mailbox.write(_mailbox);
+        let mailbox = IMailboxDispatcher { contract_address: _mailbox };
+        let local_domain = mailbox.get_local_domain();
+        self.local_domain.write(local_domain);
+        self.ownable.initializer(_owner);
+    }
+    #[abi(embed_v0)]
+    impl Upgradeable of IUpgradeable<ContractState> {
+        fn upgrade(ref self: ContractState, new_class_hash: ClassHash) {
+            self.ownable.assert_only_owner();
+            self.upgradeable._upgrade(new_class_hash);
+        }
+    }
+
+
+    #[abi(embed_v0)]
+    impl IMailboxClientImpl of IMailboxClient<ContractState> {
```

Comment:
I can't see the `assert_only_mailbox` modifier ?

Reply:
The assertion assert_only_mailbox is not used in the MailboxClient implementation, so no real need to add it for this contract.